source/common/upstream/health_checker_base_impl.cc

```diff
@@ -419,6 +419,31 @@ HealthCheckerImplBase::ActiveHealthCheckSession::clearPendingFlag(HealthTransiti
 }
 
 void HealthCheckerImplBase::ActiveHealthCheckSession::onIntervalBase() {
+  // Only if host is healthy, check whether need disable HC if there is already business traffic
+  if (parent_.disable_health_check_if_active_traffic &&
+      !host_->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC)) {
+    MonotonicTime traffic_pass_time;
+    switch (parent_.healthCheckerType()) {
+    case envoy::data::core::v3::HealthCheckerType::TCP:
+      traffic_pass_time = host_->lastTrafficPassTime();
+      break;
+    case envoy::data::core::v3::HealthCheckerType::HTTP:
+      traffic_pass_time = host_->lastTrafficPassTime2xx();
+      break;
+    case envoy::data::core::v3::HealthCheckerType::GRPC:
+      traffic_pass_time = host_->lastTrafficPassTimeGrpc();
+      break;
+    default:
+      break;
+    }
```

Comment:
What is this going to do for cases like Redis? I think you either have to handle this for all health checkers or error out the config if it is set on one that is not supported?

Reply:
I will error out the config for custom health check type.